### PR TITLE
feat(input-time-zone): allow clearing value

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #9020

## Summary

Adds `clearable` property to allow `input-time-zone` to be cleared via the UI or programmatically via `””` or `null`.

### CODE REVIEW

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 1]
```diff
@@ -77,6 +77,13 @@ export class InputTimeZone
   //
   //--------------------------------------------------------------------------
 
+  /**
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:80]: @geospatialem @DitwanP Would you mind reviewing this doc? It's based on `color-picker`'s [`clearable` prop doc](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/color-picker/color-picker.tsx#L130-L134), but I think both need some clarity tweaks.

Comment (geospatialem) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:80]: LGTM - think both use cases for `clearable` below and for `color-picker` although a bit wordy provide good context, but am flexible if @DitwanP has a recommended approach, too.

**Comment by alisonailea** [comment]
What is the expected user experience to clear the value of the InputTimeZone? I took a look at this locally and I couldn't clear values as I expected. I.E. similar to clearing values in Combox.

**Review by driskull — APPROVED** [review]
LGTM!

**Comment by jcfranco** [comment]
> What is the expected user experience to clear the value of the InputTimeZone? I took a look at this locally and I couldn't clear values as I expected. I.E. similar to clearing values in Combox.

@alisonailea [This story](https://6266d45509d7eb004aa254fb-savsemwumb.chromatic.com/?path=/story/components-controls-inputtimezone--clearable) should allow you to clear the value via the clear affordance, deselecting the active item.
